spanner/google/cloud/spanner_v1/keyset.py

```diff
@@ -60,17 +60,19 @@ def to_pb(self):
         """
         kwargs = {}
 
-        if self.start_open:
+        if self.start_open is not None:
             kwargs['start_open'] = _make_list_value_pb(self.start_open)
-
-        if self.start_closed:
+        elif self.start_closed is not None:
             kwargs['start_closed'] = _make_list_value_pb(self.start_closed)
+        else:
+            kwargs['start_closed'] = _make_list_value_pb([])
```

Comment:
As in #4515, I don't think the `if -> elif` change or the `else:` changes are warranted. (FWIW the `elif` is effectively re-doing verification from the constructor, so it's just a DRY to avoid it, the `else:` is the real problem).

Reply:
A summary of ongoing discussion in chat.

These hacks **seem** necessary because the **server** [requires][1] the start key

> The start key must be provided. It can be either closed or open.

and [requires][2] the end key

> The end key must be provided. It can be either closed or open.

I propose that we instead move that validation into the constructor, rather than do it here. This method should just be a "dumb" marshaling of already verified data.

[1]: https://github.com/googleapis/googleapis/blob/c5d0b08ecd3049c378972d833a49df99f779f97f/google/spanner/v1/keys.proto#L118
[2]: https://github.com/googleapis/googleapis/blob/c5d0b08ecd3049c378972d833a49df99f779f97f/google/spanner/v1/keys.proto#L129